opm/simulators/wells/StandardWellV_impl.hpp

```diff
@@ -2265,12 +2273,42 @@ namespace Opm
     void
     StandardWellV<TypeTag>::
     computeWellRatesWithBhp(const Simulator& ebosSimulator,
-                            const EvalWell& bhp,
+                            const std::vector<Scalar>& B_avg,
+                            const double& bhp,
+                            const bool iterate,
                             std::vector<double>& well_flux,
-                            Opm::DeferredLogger& deferred_logger) const
+                            Opm::DeferredLogger& deferred_logger)
     {
+
         const int np = number_of_phases_;
         well_flux.resize(np, 0.0);
+        WellControls* wc = well_controls_;
+        const int bhp_index = Base::getControlIndex(BHP);
+        const double orig_bhp = well_controls_iget_target(wc, bhp_index);
+        const auto orig_current = well_controls_get_current(wc);
+
+        well_controls_iset_target(wc, bhp_index, bhp);
+        well_controls_set_current(wc, bhp_index);
+
+        // iterate to get a more accurate well density
+        if (iterate) {
```

Comment:
I understand the problem better now.

I do not have access to reference result. It is possible for well potential calculation, it needs to solve the well equation first, to update the perforation pressure drops to be proper first. 

And also, at the same time, for some other purposes, it might not need it, because the perforation pressure drops are treated in an explicit manner.  There are specified perforation pressure drops there already. 

Maybe we should split them `computeWellPotentialsWithBhp` and you call `computeWellRatesWithBhp` within the previous function. 

Will this make the interface change of `B_avg` less, instead of using a boolean input arguments to put them together?

Reply:
But now I have done the work. I really don't want to spend another half day changing it all back again.

Reply:
Sure. It is fine. I put it on my list. I will handle it.